src/test/java/seedu/address/logic/commands/DeletePolicyCommandTest.java

```diff
@@ -0,0 +1,125 @@
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+
+        // the model contains the same client that contains same policy list so naturally deletes the policy
+        assertCommandSuccess(deletePolicyCommand, model, expectedMessage, expectedModel);
+    }
+
+
+    @Test
+    public void execute_invalidClientIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredClientList().size() + 1);
+        DeletePolicyCommand deletePolicyCommand = new DeletePolicyCommand(outOfBoundIndex, INDEX_FIRST_POLICY);
+
+        assertCommandFailure(deletePolicyCommand, model, MESSAGE_INVALID_CLIENT_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_invalidPolicyIndexUnfilteredList_throwsCommandException() {
+        Client client = model.getFilteredClientList().get(INDEX_FIRST_CLIENT.getZeroBased());
+        Index outOfBoundIndex = Index.fromOneBased(client.getFilteredPolicyList().size() + 1);
+        DeletePolicyCommand deletePolicyCommand = new DeletePolicyCommand(INDEX_FIRST_CLIENT, outOfBoundIndex);
+
+        assertCommandFailure(deletePolicyCommand, model, MESSAGE_INVALID_POLICY_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_validIndexFilteredList_success() {
+        showClientAtIndex(model, INDEX_FIRST_CLIENT);
+        Client client = model.getFilteredClientList().get(INDEX_FIRST_CLIENT.getZeroBased());
+        Policy policyToDelete = client.getFilteredPolicyList().get(INDEX_FIRST_POLICY.getZeroBased());
+
+        DeletePolicyCommand deletePolicyCommand = new DeletePolicyCommand(INDEX_FIRST_CLIENT, INDEX_FIRST_POLICY);
+
+        String expectedMessage = String.format(
+                DeletePolicyCommand.MESSAGE_DELETE_POLICY_SUCCESS, policyToDelete.toString()
+        ) + " from: " + client.getName().toString();
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.updateFilteredClientList(p -> p.equals(client));
+        // the model contains the same client that contains same policy list so naturally deletes the policy
+        assertCommandSuccess(deletePolicyCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidClientIndexFilteredList_throwsCommandException() {
+        showClientAtIndex(model, INDEX_FIRST_CLIENT);
+
+        Index outOfBoundIndex = INDEX_SECOND_CLIENT;
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getClientList().size());
+
+        DeletePolicyCommand deletePolicyCommand = new DeletePolicyCommand(outOfBoundIndex, INDEX_FIRST_POLICY);
+
+        assertCommandFailure(deletePolicyCommand, model, Messages.MESSAGE_INVALID_CLIENT_DISPLAYED_INDEX);
+    }
+
+    /*
+    Intentions on making test cases for filtered policy list, but am not sure how filtered list works in
+    client at the moment.
+     */
```

Comment:
I believe the filtered list just creates a list with an observable list and a given predicate. See the constructor:
```
public FilteredList(@NamedArg("source") ObservableList<E> source, @NamedArg("predicate") Predicate<? super E> predicate) {
        super(source);
        filtered = new int[source.size() * 3 / 2  + 1];
        if (predicate != null) {
            setPredicate(predicate);
        } else {
            for (size = 0; size < source.size(); size++) {
                filtered[size] = size;
            }
        }
    }
```

In your test you can just create a FilteredList stub and add it to the ModelManager stub, from there should be simple to script out test cases.